### PR TITLE
Fix name resolution in istioctl command

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/43691

Should use the inferred namespace https://github.com/hanxiaop/istio/blob/b247ddba7aeb041b4fc6e47256e15323f7e4be4e/istioctl/pkg/util/handlers/handlers.go#L140 instead of the default namespace.

### CODE REVIEW

File: releasenotes/notes/43709.yaml
Comment (ericvn) [inline releasenotes/notes/43709.yaml:5]: Can we get the issue noted here as well (https://github.com/istio/istio/issues/43691).

Comment (ericvn) [reply releasenotes/notes/43709.yaml:5]: Also, RMs should looking at release notes for anything cherry-picked since that tells them something was changed instead of a new feature which may not necessarily have a RN.

File: releasenotes/notes/43709.yaml
Comment (dhawton) [inline releasenotes/notes/43709.yaml:6]:
```suggestion
    **Fixed** resource namespace resolution for `istioctl` commands.
```

**Review by howardjohn — COMMENTED** [review]
can we add tests?

File: releasenotes/notes/43709.yaml
Comment (hanxiaop) [reply releasenotes/notes/43709.yaml:5]: I see, thanks

File: releasenotes/notes/43709.yaml
Comment (hanxiaop) [reply releasenotes/notes/43709.yaml]: Fixed, thanks

**Comment by johnma14** [comment]
Is this close to getting merged? Would like to see this in 1.17 branch. Thanks.

**Comment by hanxiaop** [comment]
> Is this close to getting merged? Would like to see this in 1.17 branch. Thanks.

@johnma14 This is waiting for a review from @istio/wg-user-experience-maintainers . Can anyone take a look? Thanks

File: istioctl/pkg/util/handlers/handlers.go
Comment (zhlsunshine) [inline istioctl/pkg/util/handlers/handlers.go:138]: What's the benefit to define this proto function?

Comment (hanxiaop) [reply istioctl/pkg/util/handlers/handlers.go:138]: This is only for testing purposes. There's no existing test that covers this, and without defining the function, it's hard to write a test for the entire function.

Comment (zirain) [reply istioctl/pkg/util/handlers/handlers.go:138]: add comment like https://github.com/istio/istio/blob/98c4f535dea16ddfa1fe04fe8a28c8f510edd0a1/pilot/pkg/networking/core/v1alpha3/tracing.go#L58

Comment (zhlsunshine) [reply istioctl/pkg/util/handlers/handlers.go:138]: okay, got it. Please add the comment for it.

Comment (hanxiaop) [reply istioctl/pkg/util/handlers/handlers.go:139]: Thanks, done.

**Review by zhlsunshine — APPROVED** [review]
LGTM

**Pull request merged**
